### PR TITLE
cli: template path format change

### PR DESCRIPTION
**Pull request by dinosk**
* FIXES path format for templates to support dot notation 

Signed-off-by: Dinos Kousidis <konstantinos.kousidis@cern.ch>

### CODE REVIEW

File: invenio_search/ext.py [hunk 1]
```diff
@@ -111,34 +112,38 @@ def _walk_dir(aliases, *parts):
...
 
-    def register_templates(self, directory, package_name):
+    def register_templates(self, directory):
         """Register templates from the provided directory.
 
         :param directory: The templates directory.
         :param package_name: The package name.
         """
         result = {}
+        module_name, parts = directory.split('.')[0], directory.split('.')[1:]
```
Comment (nharraud) [inline invenio_search/ext.py]: The `directory.split('.')` can be done only once.

File: invenio_search/ext.py [hunk 2]
```diff
@@ -111,34 +112,38 @@ def _walk_dir(aliases, *parts):
...
+        parts = tuple(parts)
 
-        def _walk_dir(*parts):
+        def _walk_dir(parts):
             resource_name = os.path.join(*parts)
 
-            for filename in resource_listdir(package_name, resource_name):
+            for filename in resource_listdir(module_name, resource_name):
+                if ES_VERSION[0] == 2 and filename.startswith('v5'):
```
Comment (nharraud) [inline invenio_search/ext.py]: This needs to be documented. If I understand correctly the function lists mappings for all versions and the wrong versions get filtered out.

File: tests/test_cli.py [hunk 3]
```diff
@@ -55,10 +56,13 @@ def test_init(app, template_entrypoints):
...
-        assert 'record_view_v1' in search.templates
-        assert 'subdirectory-file_download_v1' in search.templates
-        assert len(search.templates.keys()) == 3
+        assert 'file-download-v1' in search.templates
+        assert 'record-view-v1' in search.templates
+        assert 'subdirectory-file-download-v1' in search.templates
+        if ES_VERSION[0] == 2:
+            assert len(search.templates.keys()) == 3
+        elif ES_VERSION[0] == 5:
+            assert len(search.templates.keys()) == 4
```
Comment (nharraud) [inline tests/test_cli.py]: There should be only one template loaded in this case.

File: tests/test_invenio_search.py [hunk 4]
```diff
@@ -131,4 +132,7 @@ class ep(object):
 
     with patch('invenio_search.ext.iter_entry_points',
                return_value=template_entrypoints('invenio_search.templates')):
-        assert len(ext.templates.keys()) == 3
+        if ES_VERSION[0] == 2:
+            assert len(ext.templates.keys()) == 3
+        elif ES_VERSION[0] == 5:
+            assert len(ext.templates.keys()) == 4
```
Comment (nharraud) [inline tests/test_invenio_search.py]: Same as above. 1 template for es5

**Comment by nharraud** [comment]
This PR has some conflicts.

**Comment by dinosk** [comment]
@nharraud I resolved the conflicts, should be ready

File: invenio_search/__init__.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015, 2016, 2017 CERN.
+# Copyright (C) 2017 CERN.
```
Comment (nharraud) [inline invenio_search/__init__.py]: ???

**Comment by nharraud** [comment]
Looks good, I'll just fix the copyright year and merge.

**Pull request merged**